Content.Client/Silicons/Laws/Ui/LawDisplay.xaml.cs

```diff
@@ -80,9 +91,21 @@ public LawDisplay(EntityUid uid, SiliconLaw law, HashSet<string>? radioChannels)
                     default:
                         _chatManager.SendMessage($"{SharedChatSystem.RadioChannelPrefix}{radioChannelProto.KeyCode} {lawIdentifierPlaintext}: {lawDescriptionPlaintext}", ChatSelectChannel.Radio); break;
                 }
+                _nextAllowedPress[radioChannelButton] = _timing.CurTime + PressCooldown;
             };
 
             LawAnnouncementButtons.AddChild(radioChannelButton);
         }
     }
+
+    protected override void FrameUpdate(FrameEventArgs args)
+    {
+        base.FrameUpdate(args);
+
+        var curTime = _timing.CurTime;
+        foreach (var (button, nextPress) in _nextAllowedPress)
```

Comment:
Most of the time, most buttons are going to be enabled; so, this loop will iterate through every button once per frame and do nothing. I think it would be better from a performance perspective to store only the buttons which are currently on cooldown. You could store the (button, timespan) in an ordered queue, which would minimize the amount of work done in this function.